lib/mayaUsd/commands/baseExportCommand.cpp

```diff
@@ -231,6 +231,8 @@ MStatus MayaUSDExportCommand::doIt(const MArgList& args)
             if (fileName.empty()) {
                 fileName = tmpVal.asChar();
             }
+
+            userArgs["fileName"] = fileName;
```

Comment:
It feels a bit weird to be stuffing a (likely) processed string value back into `userArgs`, since it may very well be different from what the user actually specified. The other wrinkle here is that we know we have some work to do to properly support USD's `Ar` path resolvers, and we're baking in a resolved path that uses Maya's resolver.

What do you think of adding a `GetResolvedFileName()` method to `UsdMayaJobExportArgs` that does this resolution and returns the result? Something like:

```C++
std::string UsdMayaJobExportArgs::GetResolvedFileName() // declared const in the header
{
    MFileObject fileObj;
    fileObj.setRawFullName(fileName);

    // Make sure it's an absolute path
    fileObj.setRawFullName(fileObj.resolvedFullName());
    const std::string resolvedFileName = absoluteFile.resolvedFullName().asChar();

    if (!resolvedFileName.empty()) {
        return resolvedFileName;
    }

    return fileName;
}
```

We could then replace the `if` block with just this:

```C++
if (argData.isFlagSet(kFileFlag)) {
    fileName = userArgs.GetResolvedFileName();
}
```

And we would also call `GetResolvedFileName()` in the file texture writer. That way we have the original, unmodified `fileName` argument available if we need it for something, and we also have the path resolution stuff separated out so we can revisit it when we get to the `Ar` work.

Reply:
Yeah that's a fair call. I figured it would be representative of the users fileName argument to the exporter, which is why I was putting it into the userArgs, but you're right that it is mutated so isn't exactly 1:1.

I'll make the necessary changes this week and update the PR

Reply:
Updated the PR with the changes.
The UsdMayaJobExportArgs constructors now take fileName as an argument to instantiate it, and added a GetResolvedFileName method to return the resolved file name to the caller. 

I believe that should address the concerns you raised, but let me know if there's anything else I missed.

Reply:
Cool, that's looking better already.

I think I was imagining though that we'd keep your original changes to jobArgs.h and jobArgs.cpp that add `fileName` as a public token and member of the struct. That way, the default dictionary would have a `fileName` key in it so that when we call `GetDictionaryFromArgDatabase()`, the returned `userArgs` dictionary has that key and its value from the Maya args in it:
https://github.com/Autodesk/maya-usd/blob/dcd4094f5353d6be96fb7b61b17a41686b58ad5a/lib/mayaUsd/commands/baseExportCommand.cpp#L208

We then don't need to plumb the `fileName` down through the `UsdMayaJobExportArgs` functions separately:
https://github.com/Autodesk/maya-usd/blob/dcd4094f5353d6be96fb7b61b17a41686b58ad5a/lib/mayaUsd/commands/baseExportCommand.cpp#L288

We'd basically be treating the `fileName` arg the same as we do every other arg that can be specified to the command, and just have an additional function on `UsdMayaJobExportArgs` that processes it into a resolved value. With the most recent changes, nothing has access to the unresolved `fileName` value as far as I can tell?

Reply:
Ah I see what you're saying. I'll have to figure out how to get the file attribute working in GetDictionaryFromArgDatabase. Since I was setting it externally to the userArgs earlier, it seems like it never actually was getting set up here.

Reply:
Okay pushed a new update. I've renamed the `fileName` token to `file` to stay close to the input arguments, and have that publicly available now.

Reply:
Ahh, sorry for the confusion. I missed the `fileName`/`file` disconnect there. But yeah, it's looking great to me now. Thanks for doing that!